app/src/main/java/com/example/thecocktailsapp/utils/UIState.kt

```diff
@@ -0,0 +1,7 @@
+package com.example.starwarsmvvm.utils
+
+sealed class UIState<out T> {
+    object LOADING : UIState<Nothing>()
+    data class SUCCESS<T>(val response: T) : UIState<T>()
```

Comment:
Do we need this to be generic? Could we use Domain class instead?

Reply:
Isn't generic good as well?

Reply:
Generics are useful when. you have different types of data coming. in the network calls, if there is no need you can use the Domain